inbox/util/concurrency.py

```diff
@@ -2,24 +2,30 @@
 import random
 
 import gevent
+from backports import ssl
+from gevent import socket
+from redis import TimeoutError
 
 from nylas.logging import get_logger
 from nylas.logging.sentry import log_uncaught_errors
 log = get_logger()
+
 BACKOFF_DELAY = 30  # seconds to wait before retrying after a failure
+TRANSIENT_NETWORK_ERRS = (socket.timeout, TimeoutError, socket.error, ssl.SSLError)
```

Comment:
Nitpick: I'm not sure that `ssl.SSLError` is always transient (http://stackoverflow.com/questions/10667960/python-requests-throwing-up-sslerror)